xml/admin_ceph_gateway.xml

```diff
@@ -3279,365 +3211,211 @@ rgw_zone=us-east-1</screen>
+&prompt.cephuser.ogw;radosgw-admin period update --commit</screen>
+      <para>This generates a new period, and the &ogw; instance(s)
+        in zone us-west sends this period to other zones.</para>
+    </sect3>
+   </sect2>
+   <sect2 xml:id="ceph-rgw-failover-dr">
+     <title>Failover and Disaster Recovery</title>
+     <para>If the master zone should fail, failover to the secondary
+       zone for disaster recovery.</para>
+     <procedure>
+       <step>
+         <para>Make the secondary zone the master and default zone.
+           For example:</para>
+<screen>&prompt.cephuser;radosgw-admin zone modify --rgw-zone=<replaceable>ZONE-NAME</replaceable> --master --default</screen>
+         <para>By default, &cogw; runs in an active-active
+           configuration. If the cluster was configured to run in an active-passive
+           configuration, the secondary zone is a read-only zone. Remove the
+           <option>--read-only</option> status to allow the zone to receive
+           write operations. For example:</para>
+<screen>&prompt.cephuser;radosgw-admin zone modify --rgw-zone=<replaceable>ZONE-NAME</replaceable> --master --default \
+                                                   --read-only=false
+</screen>
+       </step>
+       <step>
+         <para>Update the period to make the changes take effect:</para>
+<screen>&prompt.cephuser;radosgw-admin period update --commit</screen>
+       </step>
+       <step>
+         <para>Restart the &cogw;:</para>
+<screen>&prompt.cephuser;systemctl restart ceph-radosgw@rgw.`hostname -s`</screen>
+       </step>
+     </procedure>
+     <para>If the former master zone recovers, revert the operation.</para>
+     <procedure>
+       <step>
+         <para>From the recovered zone, pull the latest realm configuration
+           from the current master zone.</para>
+<screen>&prompt.cephuser;radosgw-admin realm pull --url=<replaceable>URL-TO-MASTER-ZONE-GATEWAY</replaceable> \
+                           --access-key=<replaceable>ACCESS-KEY</replaceable> --secret=<replaceable>SECRET</replaceable>
+</screen>
+       </step>
+       <step>
+         <para>Make the recovered zone the master and default zone:</para>
+<screen>&prompt.cephuser;radosgw-admin zone modify --rgw-zone=<replaceable>ZONE-NAME</replaceable> --master --default</screen>
+       </step>
+       <step>
+         <para>Update the period to make the changes take effect:</para>
+<screen>&prompt.cephuser;radosgw-admin period update --commit</screen>
+       </step>
+       <step>
+         <para>Restart the &cogw; in the recovered zone:</para>
+<screen>&prompt.cephuser.ogw;systemctl restart ceph-radosgw@rgw.`hostname -s`</screen>
+       </step>
+       <step>
+         <para>If the secondary zone needs to be a read-only configuration, update the secondary zone:</para>
+<screen>&prompt.cephuser;radosgw-admin zone modify --rgw-zone=<replaceable>ZONE-NAME</replaceable> --read-only</screen>
```

Comment:
Do we need to pull the period data from the master zone here, or will doing thing and doing a period update deal with everything if we happen to have promoted the recovered but the new period/state hasn't been replicated out.

This is more of a RGW management question for Abhishek, may not have any relevance to this per say.

Reply:
@theanalyst ?

Reply:
realm pull done 2 steps above should actually have the pulled period info, period update doesn't do a pull-push, as it is essential to do a push only when we get into wierd configuration states.

Reply:
Okay, so no change required?

Reply:
yes, that was an answer to Matt's question :)